### PR TITLE
Add linting doc

### PR DESCRIPTION
**Pull request by stone-z**
Create reference for linting tool behavior and configuration

### CODE REVIEW

File: go/general_style_guide.md [hunk 1]
```diff
@@ -7,3 +7,25 @@
...
+
+### False Positives
+
+The linter's job is to enforce conformity, but if you find that one of the rules conflicts with Giant Swarm style or functional needs, it is possible to exclude the offending rule.
+
+1. For a rule which should be suppressed globally, it can be excluded with `-e < specific regex to match the error>`
+2. For a rule which should only be suppressed for one file or folder path, it can be excluded with `--skip-dirs <regex to match directories>`
+or `--skip-files <regex to match files>`
+3. For a rule which should be suppressed only for a specific code line or block, it can be silenced with `//no-lint:<linter name>`.
+This should be used sparingly to avoid forming a bad habit of simply silencing bad code.
```
Comment (kopiczko) [inline go/general_style_guide.md]: We should make it clear that we need to update the orb in that case.

File: go/general_style_guide.md [hunk 2]
```diff
@@ -7,3 +7,25 @@
 It is important to follow this guide. It makes review process easier and
 faster. It also makes our codebase consistent, easier to navigate, and
 understand.
+
+## Code Quality
+
+To prevent shipping problematic or unsafe code, [`architect-orb`](https://github.com/giantswarm/architect-orb)'s [`go-test`](https://github.com/giantswarm/architect-orb/blob/master/src/commands/go-test.yaml) and [`go-test-legacy`](https://github.com/giantswarm/architect-orb/blob/master/src/commands/go-test-legacy.yaml) commands include running several linting tools via [`golangci-lint`](https://github.com/golangci/golangci-lint).
```
Comment (kopiczko) [inline go/general_style_guide.md]: Can you wrap the lines like in the rest of this page?

File: go/general_style_guide.md (hunk 1 above)
Comment (stone-z) [reply go/general_style_guide.md]: Good idea, added

File: go/general_style_guide.md (hunk 2 above)
Comment (stone-z) [reply go/general_style_guide.md]: :heavy_check_mark:

File: go/general_style_guide.md (hunk 1 above)
Comment (kopiczko) [reply go/general_style_guide.md]: I mean in case of "global" false positive.

Comment (stone-z) [reply go/general_style_guide.md]: Options 1 and 2 are both global, option 3 is per-line

**Review by TheoBrigitte — APPROVED** [review]
Nice

File: go/general_style_guide.md (hunk 1 above)
Comment (stone-z) [reply go/general_style_guide.md]: I clarified Option 2 which I think helps -- feel free to suggest if you have something else in mind

File: go/general_style_guide.md [hunk 3]
```diff
@@ -7,3 +7,30 @@
...
+
+The linter's job is to enforce conformity, but if you find that one of the rules conflicts with Giant Swarm style or functional needs, it is possible to exclude the offending rule.
+
+1. For a specific rule which should be suppressed globally, it can be excluded with `-e < specific regex to match the error>`
+2. For a folder or file path which should never be linted (globally), it can be excluded with `--skip-dirs <regex to match directories>`
+or `--skip-files <regex to match files>`
+3. For a rule which should be suppressed only for a specific code line or block, it can be silenced in the code with `//no-lint:<linter name>`.
+This should be used sparingly to avoid forming a bad habit of simply silencing bad code.
+
+Options 1 and 2 can only be applied by updating the command in `architect-orb`.
```
Comment (kopiczko) [inline go/general_style_guide.md]:
```suggestion
**NOTE:** Options 1 and 2 can only be applied by updating the command in `architect-orb`.
```

**Review by kopiczko — APPROVED** [review]
Lines are still not wrapped for the "False Positives" section. Otherwise LGTM.

File: go/general_style_guide.md (hunk 3 above)
Comment (stone-z) [reply go/general_style_guide.md]: Wrapped and added a note

**Review by ubergesundheit — APPROVED** [review]

**Pull request merged**